templates/smaily-woocommerce-admin.php

```diff
@@ -356,6 +364,83 @@ class="smaily-toggle"
 			</table>
 		</div>
 
+		<div id="checkout_subscribe">
+			<table class="form-table">
+				<tbody>
+					<tr class="form-field">
+						<th scope="row">
+							<label for="checkbox_display">
+							<?php echo esc_html__( 'Display', 'smaily' ); ?>
+							</label>
+						</th>
+						<td id="smaily_checkout_display_visibility">
+							<?php
+							$checkbox_display_values = array(
+								'visible_unchecked' => __( 'Visible unchecked', 'smaily' ),
+								'visible_checked'   => __( 'Visible checked', 'smaily' ),
+								'hidden_unchecked'  => __( 'Hidden unchecked', 'smaily' ),
+
+							);
+							foreach ( $checkbox_display_values as $dp_value => $dp_translation ) {
+								$checked = $cb_display_selected === $dp_value ? 'checked' : '';
+
+								$input  = '<label for="checkbox_' . $dp_value . '">';
+								$input .= '<input type="radio" ';
+								$input .= 'id="checkbox_' . $dp_value . '" ';
+								$input .= 'name="checkbox_display" ';
+								$input .= 'value="' . $dp_value . '" ' . $checked . '>';
+								$input .= $dp_translation . '</label>';
+								echo( $input );
+							}
+							?>
+						</td>
+					</tr>
+					<tr class="form-field">
+						<th scope="row">
+							<label for="checkbox_order">
+							<?php echo esc_html__( 'Before / After', 'smaily' ); ?>
+							</label>
+						</th>
+						<td>
+							<select id="cb-before-after" name="checkbox_order">
+								<option value="before" <?php echo( 'before' === $cb_order_selected ? 'selected' : '' ); ?> >
+									<?php echo esc_html__( 'Before', 'smaily' ); ?>
+								</option>
+							<option value="after" <?php echo( 'after' === $cb_order_selected ? 'selected' : '' ); ?>>
+								<?php echo esc_html__( 'After', 'smaily' ); ?>
+							</option>
+							</select>
+						</td>
+					</tr>
+					<tr class="form-field">
+						<th scope="row">
+							<label for="checkbox_location">
+							<?php echo esc_html__( 'Display location', 'smaily' ); ?>
+							</label>
+						</th>
+						<td>
+							<select id="checkbox-location" name="checkbox_location">
+							<?php
+							$cb_loc_available = array(
+								'customer_details'       => __( 'Customer details', 'smaily' ),
+								'checkout_billing_form'  => __( 'Billing form', 'smaily' ),
+								'checkout_shipping_form' => __( 'Shipping form', 'smaily' ),
+								'checkout_registration_form' => __( 'Registration form', 'smaily' ),
+							);
+							// Display option and select saved value.
+							foreach ( $cb_loc_available as $loc_value => $loc_translation ) {
+								$selected = $cb_loc_selected === $loc_value ? 'selected' : '';
+								echo( "<option value='$loc_value' $selected>$loc_translation</option>" );
+							}
+							?>
+							</select>
```

Comment:
The idea for providing preselect for values is that I think it makes our plugin more user friendly than just providing text box to enter hook where to add the checkbox. Most of these hooks on [WooCommerce hooks page](https://docs.woocommerce.com/wc-apidocs/hook-docs.html) are not for this use-case. 
This checkbox has to be inside the form element of the checkout page and this should be checked by reading source code - not many admins capable of doing. MailChimp has a different solution by providing a text box, but I think this one is better. 
Also, there is only couple of values where the join newsletter checkbox would make sense adding.

Reply:
There is also a [visual guide](https://www.tychesoftwares.com/woocommerce-checkout-page-hooks-visual-guide-with-code-snippets/) for checkout page hooks, but this is not an official site so I wouldn't link it in the description or help text for the option. I didn't find anything similar in WooCommerce docs.

Reply:
Yes, providing concrete selection options is better for less tech-savvy users, but it doesn't take a prophet to say that it does not cover all edge cases. For sure there will be somebody who wants more :)

But I digress. Let's leave it as a select box now.

PS. I guess Mailchimp's approach was to be ultimately flexible, i.e. you could target even a 3rd party plugin of WooCommerce that extends checkout forms.